crates/test-programs/tests/http_tests/runtime/wasi_http_tests.rs

```diff
@@ -4,11 +4,57 @@ use wasmtime::{Config, Engine, Linker, Module, Store};
 use wasmtime_wasi::{sync::WasiCtxBuilder, WasiCtx};
 use wasmtime_wasi_http::WasiHttp;
 
+use http_body_util::{BodyExt, Full};
+use hyper::server::conn::http1;
+use hyper::{body::Bytes, service::service_fn, Request, Response};
+use std::{error::Error, net::SocketAddr};
+use tokio::net::TcpListener;
+
+async fn test(req: Request<hyper::body::Incoming>) -> http::Result<Response<Full<Bytes>>> {
+    let method = req.method().to_string();
+    let body = req.collect().await.unwrap().to_bytes();
+    Response::builder()
+        .status(http::StatusCode::OK)
+        .header("x-wasmtime-test-method", method)
+        .body(Full::new(Bytes::from(body)))
+}
+
+async fn async_run_serve() -> Result<(), Box<dyn Error + Send + Sync>> {
+    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
+
+    let listener = TcpListener::bind(addr).await?;
+
+    loop {
+        let (stream, _) = listener.accept().await?;
+
+        tokio::task::spawn(async move {
+            if let Err(err) = http1::Builder::new()
+                .serve_connection(stream, service_fn(test))
+                .await
+            {
+                println!("Error serving connection: {:?}", err);
+            }
+        });
+    }
+}
+
+fn run_server() -> Result<(), Box<dyn Error + Send + Sync>> {
+    let rt = tokio::runtime::Runtime::new()?;
+    let _ent = rt.enter();
+
+    rt.block_on(async_run_serve())?;
+    Ok(())
+}
+
 pub fn instantiate_inherit_stdio(
     data: &[u8],
     bin_name: &str,
     _workspace: Option<&Path>,
 ) -> anyhow::Result<()> {
+    let _thread = std::thread::spawn(|| {
+        run_server().unwrap();
+    });
```

Comment:
This is going to break as soon as we run more than one test in a process, which I hope we will do soon. I'll work on the structural changes to the test suite to work that out soon, but its ok for now